src/test/kotlin/org/rekotlin/TestFakes.kt

```diff
@@ -114,6 +114,38 @@ internal class TestStoreSubscriber<T> : StoreSubscriber<T> {
     }
 }
 
+/**
+ * A subscriber contains another sub-subscribers [StoreSubscriber], which could be subscribe/unsubscribe at some point
+ */
+internal class ViewSubscriberTypeA(var store: Store<TestStringAppState>) : StoreSubscriber<TestStringAppState> {
+    private val viewB by lazy { ViewSubscriberTypeB(store) }
+    private val viewC by lazy { ViewSubscriberTypeC() }
+
+    override fun newState(state: TestStringAppState) {
+        when(state.testValue){
```

Comment:
I am not sure why to subscribe or unsubscribe in a newState function.
A class should subscribe to the state when it is getting created and unsubscribe when it is getting destroyed.
Any conditional subscription or unsubscribing should be delegated to the separate class by a dispatch action that triggers that class creation.

@jitinsharma, @kaustavjaiswal -- I need your view before merging this.

Anyway, in meanwhile @NemoOudeis Could you please add to the README.md, that number of concurrent writes should be less than the number of concurrent reads in the subscriptions list.

Reply:
I have also doubts on scenarios where a subscription is required on new state function.

Reply:
@NemoOudeis Would it be possible for you to mention an example of real-world usage (if possible) within your application where such behavior is required?

Reply:
I can describe an example, the code is closed source, so I cannot share it here.

We use a component (think react components, i.e. UI with some behavior) hierarchy that assembles the UI for the user, an example of such a hierarchy could be:
```
     +------- Screen Component -------+
     |               |                |
     |               |                |
Component A          |            Component B
                     |
                Compontent C
```

Each of the components subscribes to (sub-)states and behaves accordingly.
In cases where the hierarchy is known at compile time it's easy, we can just create and the components A, B and C and subscribe them at our own convenience.

The problem occurs when we don't know the hierarchy at compile time, e.g. a components should be shown (or not) based on some data fetched from a server. In that case the data is fetched -> send to the store -> reducers update state -> store notifies subscribers. In that notification (`newState) the `Screen Component` decides to attach (or detach) another component - which requires that component to subscribe unsubscribe (this happens in the call stack of `newState`).

Does that make sense?